note.h

```diff
@@ -25,28 +25,43 @@
 #define NOTE_C_VERSION_MINOR 1
 #define NOTE_C_VERSION_PATCH 1
 
-// Determine our basic floating data type.  In most cases "double" is the right answer, however for
-// very small microcontrollers we must use single-precision.
+// If double and float are the same size, then we must be on a small MCU. Turn
+// on NOTE_LOWMEM to conserve memory.
 #if defined(FLT_MAX_EXP) && defined(DBL_MAX_EXP)
 #if (FLT_MAX_EXP == DBL_MAX_EXP)
-#define NOTE_FLOAT
+#define NOTE_LOWMEM
 #endif
 #elif defined(__FLT_MAX_EXP__) && defined(__DBL_MAX_EXP__)
 #if (__FLT_MAX_EXP__ == __DBL_MAX_EXP__)
-#define NOTE_FLOAT
+#define NOTE_LOWMEM
 #endif
 #else
 #error What are floating point exponent length symbols for this compiler?
 #endif
 
-// If using a short float, we must be on a VERY small MCU.  In this case, define additional
-// symbols that will save quite a bit of memory in the runtime image.
-#ifdef NOTE_FLOAT
+#ifndef JNUMBER
```

Comment:
The placement of this `#if` :point_up:  is not obvious. Why is it not below `#ifdef NOTE_LOWMEM`?

I think it would make it easier to understand the subsequent `#if` logic.

Reply:
The idea here was to allow the user to define `JNUMBER` themselves, if desired. Then none of our  preprocessor logic here applies. Their definition takes precedence. So I read it as "if the user hasn't specified what `JNUMBER` should be already, apply these rules."

Reply:
So if the user provides their own `JNUMBER`, then we don't need to consider if `NOTE_LOWMEM` is defined?